scripts/production/repair_mlss_tags.py

```diff
@@ -0,0 +1,117 @@
+            OR ml1.class = "ConstrainedElement.constrained_element")
+        AND (ml2.class = "GenomicAlignBlock.multiple_alignment" OR ml2.class LIKE "GenomicAlignTree.%%")
+        AND ml1.type NOT LIKE "pGERP%%"
+        AND ml2.type NOT LIKE "pEPO%%";
+    """
+}
+
+
+def repair_mlss_tag(dbc: DBConnection, mlss_tag: str) -> None:
+    """Repairs the given MLSS tag in the database, recomputing the expected values.
+
+    It uses the predefined query in `EXPECTED_VALUES_SQL` for the given MLSS tag to extract the
+    expected value for each MLSS id. It also adds any missing tags as well as removes any rows that
+    may contain an invalid MLSS id.
+
+    Args:
+        dbc: Compara database connection handler.
+        mlss_tag: MLSS tag as found in the ``method_link_species_set_tag`` table.
+
+    """
+    with dbc.connect() as connection:
+        # Get the MLSS tags in method_link_species_set_tag table
+        mlss_tag_values = connection.execute(f"SELECT method_link_species_set_id AS mlss_id, value "
+                                             f"FROM method_link_species_set_tag WHERE tag = '{mlss_tag}';")
+        mlss_tags = {row.mlss_id: row.value for row in mlss_tag_values.fetchall()}
+        # Extract the expected tag value based on the source data
+        expected_values = connection.execute(EXPECTED_VALUES_SQL[mlss_tag])
+        # Check that each tag has the correct value, fixing those that do not, and add any missing tags
+        for row in expected_values:
+            if row.mlss_id in mlss_tags:
+                # NOTE: due to internal conversions, we need to ensure both sides have the same time
+                if str(mlss_tags[row.mlss_id]) != str(row.value):
+                    connection.execute(f'UPDATE method_link_species_set_tag SET value = {row.value} '
+                                       f'WHERE method_link_species_set_id = {row.mlss_id} '
+                                       f'    AND tag = "{mlss_tag}";')
+                    print(f"Repaired MLSS tag '{mlss_tag}' for MLSS id '{row.mlss_id}'")
+                del mlss_tags[row.mlss_id]
+            else:
+                connection.execute(f'INSERT INTO method_link_species_set_tag '
+                                   f'VALUES ({row.mlss_id}, "{mlss_tag}", {row.value});')
+                print(f"Added missing MLSS tag '{mlss_tag}' for MLSS id '{row.mlss_id}'")
+        # Delete those MLSS tags that do not match any MLSS in method_link_species_set table
+        for mlss_id in mlss_tags.keys():
+            connection.execute(f'DELETE FROM method_link_species_set_tag '
+                               f'WHERE method_link_species_set_id = {mlss_id};')
+            print(f"Deleted unexpected MLSS tag '{mlss_tag}' for MLSS id '{mlss_id}'")
+
+
+if __name__ == '__main__':
+    parser = ArgumentParser(description='Repairs the requested MLSS tag(s).')
+    parser.add_argument('--url', required=True, help='URL to the Compara database')
```

Comment:
Since this is specific to compara databases and use, I wonder if we could have an option (and perhaps a util somewhere in ensembl-py) to take `--host`,  and `--db_name`. A python version of: [this](https://github.com/Ensembl/ensembl-compara/blob/release/106/modules/Bio/EnsEMBL/Compara/Utils/Registry.pm#L293-L368) (but without the involvement of aliases of course - registry stuff). Not necessary for this PR, just a thought for ease of future use in other python scripts too.

Reply:
That is actually a good idea. I think the best approach is to use [`argschema`](https://argschema.readthedocs.io/en/latest/) and introduce this kind of interface class in `ensembl-py`. Will add it to my notes and try to implement it soon.